Octokit/Http/Response.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Net;
+
+namespace Octokit.Internal
+{
+    /// <summary>
+    /// Represents a generic HTTP response
+    /// </summary>
+    public class Response : IResponse
```

Comment:
> Ideally I'd like everything under Octokit.Internal to be marked as internal - my thinking on this is that if there's a compelling case to make a type public, it shouldn't be in this namespace.

Hmm, that's not the original intent I had for this namespace. It wasn't to mark types as `internal`. The keyword suffices for that.

It's specifically for types that most people don't need to see directly, but many, if not most, need to use. For example, `ApiResponse` is in this namespace.

Most users can get by importing just the `Octokit` namespace and these other types won't "pollute" their namespace, but are still usable as return values. If they need these types directly, they can import `Octokit.Internal`.

I'm not opposed to making these specific case internal. I just wanted to clarify why we have `Octokit.Internal` in the first place. Maybe a better name for that namespace is in order. I wonder if `Octokit.Internals` is slightly better to distinguish between the keyword and the fact that these types represent the internal workings of the lib. :smile: